Google.Api.Generator/Generation/SnippetCodeGenerator.cs

```diff
@@ -184,55 +184,69 @@ private class SnippetDef
             // TODO: Reconsider this ugly hack when refactoring after removing the service-per-file snippets.
             private bool Canonical => RegionTagDisambiguation is null;
             private IEnumerable<ClientMethod.Types.Parameter> ClientMethodParameters { get; }
+            // May be null, in which case this snippet is for a single client method only.
+            private IEnumerable<ClientMethod.Types.Parameter> AdditionalClientMethodParameters { get; }
```

Comment:
I don't really understand what this would be. If a method is overloaded, there are multiple *lists* of parameters. In other words, I'd expect this to be a `List<List<Parameter>>` or equivalent.

Or is this *only* for the case where the method just varies by "does it take CallSettings or CancellationToken"?

Reply:
Okay, that's good to know. I think it's definitely worth adding a comment, along the lines of:

```csharp
// Parameters for a single additional method covered by this snippet.
// (If we ever need to cover more than two overloads in the same snippet, this can become a list of lists.)
```